examples/ct/handlers.go

```diff
@@ -283,33 +283,54 @@ func addChainInternal(ctx context.Context, c LogContext, w http.ResponseWriter,
 		return http.StatusBadRequest, fmt.Errorf("failed to verify add-chain contents: %v", err)
 	}
 
-	// Build up the SCT and MerkleTreeLeaf. The SCT will be returned to the client and
-	// the leaf will become part of the data sent to the backend.
+	// This is now.
+	timestamp := uint64(c.TimeSource.Now().UnixNano() / millisPerNano)
+
+	// Build the MerkleTreeLeaf that gets sent to the backend, and make a trillian.LogLeaf for it.
 	var issuer *x509.Certificate
 	if len(chain) > 1 {
 		issuer = chain[1]
 	}
-	merkleLeaf, sct, err := signerFn(c.logKeyManager, chain[0], issuer, c.TimeSource.Now())
+	merkleLeaf, err := makeLeafFn(chain[0], issuer, timestamp)
 	if err != nil {
-		return http.StatusInternalServerError, fmt.Errorf("failed to build SCT and Merkle leaf: %v %v", sct, err)
+		return http.StatusBadRequest, fmt.Errorf("failed to build MerkleTreeLeaf: %v", err)
 	}
-
-	// Send the Merkle tree leaf on to the Log server.
 	leaf, err := buildLogLeafForAddChain(c, *merkleLeaf, chain)
 	if err != nil {
 		return http.StatusInternalServerError, fmt.Errorf("failed to build LogLeaf: %v", err)
 	}
+
+	// Send the Merkle tree leaf on to the Log server.
 	req := trillian.QueueLeavesRequest{LogId: c.logID, Leaves: []*trillian.LogLeaf{&leaf}}
```

Comment:
I thought I was specific. I commented on the implications for preventing the storage layer being able to shard a large batch into multiple transactions and the inefficiency of giving the client back data it already just gave the server.